### PR TITLE
implement time_monitor to return score for HPO

### PR DESCRIPTION
**Pull request by eunwoosh**
This PR contains changes for HPO.

There are two changes.
1. Add _hpo_config.yaml_ into OTE model directory.
2. Implement code flow to provide model score per epoch using Callback class.

### CODE REVIEW

**Review by evgeny-izutov — APPROVED** [review]

File: mmseg/apis/ote/apis/segmentation/ote_utils.py [hunk 1]
```diff
@@ -40,6 +41,18 @@ def on_train_batch_end(self, batch, logs=None):
...
 
+    def on_epoch_end(self, epoch, logs=None):
+        self.past_epoch_duration.append(time.time() - self.start_epoch_time)
+        self._calculate_average_epoch()
+        score = None
+        if hasattr(self.update_progress_callback, 'metric') and isinstance(logs, dict):
+            score = logs.get(self.update_progress_callback.metric, None)
+
+        if score is not None:
+            self.update_progress_callback(self.get_progress(), score=float(score))
```
Comment (evgeny-izutov) [inline mmseg/apis/ote/apis/segmentation/ote_utils.py]: `int(self.get_progress())`?

Comment (eunwoosh) [reply mmseg/apis/ote/apis/segmentation/ote_utils.py]: "UpdateProgressCallback" class in ote_sdk/ote_sdk/entities/train_parameters.py expects to get float type progress value like bellow.

class UpdateProgressCallback(Protocol):
    def __call__(self, progress: float, score: Optional[float] = None):

Comment (evgeny-izutov) [reply mmseg/apis/ote/apis/segmentation/ote_utils.py]: I see, nevertheless [bellow](https://github.com/openvinotoolkit/mmsegmentation/blob/98c77a9d76e848a3a0c12d6c38cde4d33fd7d3da/mmseg/apis/ote/apis/segmentation/ote_utils.py#L54) it receives `int`. Moreover it has the [test](https://github.com/openvinotoolkit/mmsegmentation/blob/98c77a9d76e848a3a0c12d6c38cde4d33fd7d3da/tests/test_ote/test_api.py#L182) to check `int`.

Comment (eunwoosh) [reply mmseg/apis/ote/apis/segmentation/ote_utils.py]: Oh, I see. Actually our similar PR to other tasks(cls and det) had no if-else branch for invoking self.update_progress_callback.
But I found that this code has been modified now like above with this comment "Workaround for NNCF trainer, which uses callback of a different type.". I think that it's better to implement by original way now. Thanks for detailed feedback.

Comment (eunwoosh) [reply mmseg/apis/ote/apis/segmentation/ote_utils.py]: I found that inference progress_callback function and train progress_callback function expect other type of progress value. In inference, it expects int type value as you said, but in train, float type is expected. please refer [link](https://github.com/openvinotoolkit/mmsegmentation/blob/98c77a9d76e848a3a0c12d6c38cde4d33fd7d3da/tests/test_ote/test_api.py#L154).

**Pull request merged**